transport/src/main/java/io/netty/channel/DefaultChannelHandlerContext.java

```diff
@@ -351,13 +352,21 @@ public ChannelHandlerContext fireChannelRead(final Object msg) {
     private void findAndInvokeChannelRead(Object msg) {
         DefaultChannelHandlerContext ctx = findContextInbound(MASK_CHANNEL_READ);
         if (ctx == null) {
-            ReferenceCountUtil.release(msg);
+            closeOrRelease(msg);
             notifyHandlerRemovedAlready();
             return;
         }
         ctx.invokeChannelRead(msg);
     }
 
+    private static void closeOrRelease(Object msg) {
```

Comment:
I saw this method a lot of times... Consider factoring out to a static helper method.